nodejs/lancedb/embedding/openai.ts

```diff
@@ -18,19 +18,26 @@ import { EmbeddingFunction } from "./embedding_function";
 import { register } from "./registry";
 
 export type OpenAIOptions = {
-  apiKey?: string;
-  model?: string;
+  apiKey: string;
+  model:
+    | "text-embedding-ada-002"
+    | "text-embedding-3-large"
+    | "text-embedding-3-small";
```

Comment:
IIUC, each time OpenAI releases a new model, we'll have to update our package. Users will have to wait until a newer release of LanceDB to use those models. Is that correct?

Reply:
It looks like we currently have a hard dependency on openai, so users will be locked to our version regardless.